src/averageAges.js

```diff
@@ -20,6 +20,16 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const onlyMen = people.filter(men => men.sex === 'm');
+  const ages = onlyMen.map(year => year.died - year.born);
+
+  const agesCentury = onlyMen
+    .filter(men => Math.ceil(men.died / 100) === century)
+    .map(year => year.died - year.born);
+
+  return typeof century === 'number'
+    ? agesCentury.reduce((sum, age) => sum + age, 0) / agesCentury.length
+    : ages.reduce((sum, age) => sum + age, 0) / ages.length;
```

Comment:
You've used `filter`, `map`, `filter`, `map` and `reduce` here, it means that you've iterated array 5 times🥲

It can be simplified just to using `filter` and `reduce` in this function 🙂 

Just use ternary operator inside filter method and select needed condition to filter people, and then use reduce to calculate average age

Reply:
Also you've used logic to find average ages in next function, so it can be moved to separate helper function